### PR TITLE
platform library are missing SAL annotation 

### LINKED ISSUE #1598: Some files in platform library are missing SAL annotation
- [ ]  `ebpf_extension_kernel.c`
- [ ]  `ebpf_handle_kernel.c`
- [ ]  `ebpf_extension_user.c`
- [ ]  `ebpf_handle_user.c`

### PR DESCRIPTION
**Pull request by delaramamiri**
## Description
This  PR addresses missing SAL annotations for the following files: 
```
 ebpf_handle_kernel.c
 ebpf_handle_user.c
```
Action items:
1. Functions listed in `ebpf_handle.h` are investigated and proper SAL annotations are added to the pointers. 
The original issue is listed in #1598  include the following files. However, these files are not present anymore after prior PRs. 
```
ebpf_extension_kernel.c
 ebpf_extension_user.c
```

## Testing

1. Clean the solution
2. rebuild the cleaned solution
3. "Analyze and code cleanup" and then "Run code analysis on Solution" using visual studio. Then warnings are investigated to see if any of them is related to the modified function to make sure the compiler enables automated static analysis tools to analyze code more accurately and it is not detecting any problem. 

## Documentation

No
closes #1598

### CODE REVIEW

File: libs/platform/ebpf_handle.h [hunk 1]
```diff
@@ -38,7 +38,7 @@ extern "C"
      *  operation.
      */
     _Must_inspect_result_ ebpf_result_t
-    ebpf_handle_create(ebpf_handle_t* handle, struct _ebpf_base_object* object);
+    ebpf_handle_create(_Out_ ebpf_handle_t* handle, _In_ struct _ebpf_base_object* object);
```
Comment (dv-msft) [inline libs/platform/ebpf_handle.h:41]: Shouldn't the ```_In_``` parameter here (and elsewhere in this PR) really be ```_InOut_```?

AFAIK ```_In_``` parameters _almost_ always require ```const``` as well, but in this case the ```object``` parameter _is_ indirectly modified in ```acquire_reference()```, so ```_InOut_``` is more appropriate.

Comment (dv-msft) [reply libs/platform/ebpf_handle.h:41]: Apologies for the typo.  That should be ```_Inout_```

Comment (Alan-Jowett) [reply libs/platform/ebpf_handle.h:41]: [sal-basics](https://learn.microsoft.com/en-us/cpp/code-quality/understanding-sal?view=msvc-170#sal-basics)
image

General rule is:
Const pointer implies _In_ 
Modifiable pointer implies _Inout_
If the pointer is invalid on input, then _Out_

In this case, the call to ebpf_handle_create modified the object (by incrementing its ref-count), so _Inout_ probably makes more sense.

Comment (delaramamiri) [reply libs/platform/ebpf_handle.h]: Thank you both for noting. I just updated the files.

File: libs/platform/ebpf_handle.h [hunk 2]
```diff
@@ -38,7 +38,7 @@ extern "C"
      *  operation.
      */
     _Must_inspect_result_ ebpf_result_t
-    ebpf_handle_create(ebpf_handle_t* handle, struct _ebpf_base_object* object);
+    ebpf_handle_create(_Out_ ebpf_handle_t* handle, _Inout_ struct _ebpf_base_object* object);
```
Comment (Alan-Jowett) [inline libs/platform/ebpf_handle.h:41]: Maybe update the doxygen as well to be [in,out]? See:
https://github.com/microsoft/ebpf-for-windows/blob/e7302835b1cb443d479cd6b43dfe5838ffe2db84/libs/platform/ebpf_ring_buffer.h#L48
as an example.

Comment (delaramamiri) [reply libs/platform/ebpf_handle.h:41]: Right, good point. Updated the doxygen as well.

**Review by dv-msft — DISMISSED** [review]

File: libs/platform/kernel/ebpf_handle_kernel.c [hunk 3]
```diff
@@ -18,7 +18,7 @@ ebpf_handle_table_terminate()
 {}
 
 _Must_inspect_result_ ebpf_result_t
-ebpf_handle_create(ebpf_handle_t* handle, ebpf_base_object_t* object)
+ebpf_handle_create(_Out_ ebpf_handle_t* handle,_Inout_ ebpf_base_object_t* object)
```
Comment (dthaler) [inline libs/platform/kernel/ebpf_handle_kernel.c:21]:
```suggestion
ebpf_handle_create(_Out_ ebpf_handle_t* handle, _Inout_ ebpf_base_object_t* object)
```

File: libs/platform/user/ebpf_handle_user.c [hunk 4]
```diff
@@ -40,7 +40,7 @@ ebpf_handle_table_terminate()
 }
 
 _Must_inspect_result_ ebpf_result_t
-ebpf_handle_create(ebpf_handle_t* handle, ebpf_base_object_t* object)
+ebpf_handle_create(_Out_ ebpf_handle_t* handle,_Inout_  ebpf_base_object_t* object)
```
Comment (dthaler) [inline libs/platform/user/ebpf_handle_user.c:43]:
```suggestion
ebpf_handle_create(_Out_ ebpf_handle_t* handle, _Inout_  ebpf_base_object_t* object)
```

**Review by dthaler — DISMISSED** [review]
Ok except for whitespace nits

**Review by dthaler — APPROVED** [review]

**Review by Alan-Jowett — APPROVED** [review]

**Pull request merged**